### DISCUSSION TITLE
Is it possible to connect bluetooth device which  supports bluetooth 2.1 + edr version?

### DISCUSSION THREAD

**Discussion by iamalwaysnewbie** [kenjdavidson/react-native-bluetooth-classic#129] [Q&A]
I wanna build  application which handle scanned data using bluetooth scanner.
And I am using  PM3 model made by Point Mobile manufacturer. It supports bluetooth 2.1 + edr version.
(https://pointmobile.com/en/products-bluetoothscanner/PM3)

Is it Possible to use this library?

**Comment by kenjdavidson** [comment]
Just taking a quick look - it looks like this uses the HID profile.  This library doesn't support any of the `BluetoothProfile`(s), it's solely (at this point) used for openning and communicating serially - which shouldn't be a problem as long as the device itself allows it. 

I couldn't see anything regarding API or development info in their data sheet (or anywhere else).  You can give it a shot and if you think it's half working, feel free to open a ticket / pull request if you get stuff sorted.

It doesn't look like (and I'm not bluetooth expert) that that specification wouldn't allow serial communication.

**Reply by iamalwaysnewbie** [reply]
Thanks for your reply.

**Reply by iamalwaysnewbie** [reply]
I checked this device, and It also supports SPP(Serial Port Profile) options. 
After choosing SPP option, I write code and then onDataReceived event handler is worked.
And then,  I checked that event parameter object has data property. 
**But, event.data is empty string.**
------------------------------------------
1.[Event Handler Code]
----------------------------------------------------------------
`device.onDataReceived(event => {
          console.log(event);
          console.log(event.data);
        });`

-------------------------------------------------------------------
2. [console.log(event)] result
---------------------------------------------------------------------------------------------------
`{"data": "",  "device": {"address": "00:19:01:74:54:53", "bonded": true, "extra": {}, "id": "00:19:01:74:54:53", "name": "SN2116310815"}, "timestamp": "2021-08-26T14:45:29.611Z"}`

How can I fix that?

**Reply by iamalwaysnewbie** [reply]
Oh, Sorry. I solved it
this device has three options for Blutooth Data Format-  Packet Data, Barcode Only, Barcode EOT .
anyway, i changed to [Barcode Only] option.
And then it worked.

**Comment by kenjdavidson** [comment]
Nice. Good stuff!

Get Outlook for Android<https://aka.ms/ghei36>
________________________________
